kernel/mem_domain.c

```diff
@@ -99,9 +99,10 @@ void k_mem_domain_init(struct k_mem_domain *domain, u8_t num_parts,
 
 		for (i = 0U; i < num_parts; i++) {
 			__ASSERT(parts[i] != NULL, "");
-			__ASSERT((parts[i]->start + parts[i]->size) >
-				 parts[i]->start, "");
 
```

Comment:
Questions, would this ASSERT never happens because the size would be at least 1 ?

Reply:
Correct

Reply:
Is it possible that someone might not want to use the APPMEM partition macros, and just use the k_mem_partition and k_mem_domain directly? And, then, initialize a memory domain in that way? Then we should ASSERT that the size is not zero. Or am I missing something?

Reply:
@ioannisg I'm going to get rid of these changes to mem_domain.c in the next version of this patch

The goal of the next version of this patch will be to print a comprehensible error message if a 0-size partition is defined, instead of the mess of linker errors we get if the placeholder is removed

Reply:
and this is done now.